docs/source/api/notebooks/Notebook_4_Customized_Models.ipynb

```diff
@@ -0,0 +1,456 @@
+{
+ "cells": [
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "# Notebook 4: Use GraphStorm APIs for Customizing Model Components\n",
+    "\n",
+    "This notebook provides an example about how to customize components of GML models to fit specific requirements. These customized models normally extend GraphStorm higher-level APIs, enabling them to not only implement their own functionalities, but also to easily integrate into GraphStorm training and inference pipelines.\n",
+    "\n",
+    "----\n",
+    "\n",
+    "### An Example of a Customized Model\n",
+    "\n",
+    "A widely used GNN model is the `RGAT` model, proposed by [Relational Graph Attention Networks](https://arxiv.org/abs/1904.05811). The original `RGAT` model considers the different importance of neighbors for a node and leverages attention mechanism to aggregate messages from neighbors within the same relation type. Then, aggregations of neighbors from different relation types are added together as the output representations of a node.\n",
+    "\n",
+    "$h_i = \\sum_{r\\in \\mathscr{R}}\\sum_{j\\in \\mathcal{N}^{r}_{(i)}} \\alpha^{r}_{i,j} W^{r} h_j^{r}$\n",
+    "\n",
+    "An alternative way to aggregate representations across different relation types is to use attention instead of summation. We can use an additional weight set to compute the attention coefficients for different relation types, and then compute the weighted sum of aggregations across relation types.\n",
```

Comment:
```suggestion
    "An alternative way to aggregate representations across different relation types is to use attention instead of summation. We can use an additional weight set to compute the attention coefficients for different relation types,\n",
    "$$\\beta^r_i = \\dfrac{exp(h^{r}_i \\cdot \\phi)}{\\sum_{k \\in \\{1, \\dots, \\mathscr{R}\\}} exp(h^{k}_i \\cdot \\phi)},$$\n",
    "and then compute the weighted sum of aggregations across relation types,\n",
    "$$h_i = \\sum_{r \\in \\mathscr{R}}{\\beta^r_i \\times h_i^{r}},$$\n",
    "$$h_i^{r} = LeakyReLU(\\sum_{j\\in \\mathcal{N}^{r}_{(i)}} \\alpha^{r}_{i,j} W^{r} h_j^{r}).$$\n",
    "\n",
```